csharp/ql/lib/semmle/code/csharp/File.qll

```diff
@@ -209,7 +209,12 @@ class File extends Container, @file {
    * A source file can come from a PDB and from regular extraction
    * in the same snapshot.
    */
-  predicate isPdbSourceFile() { file_extraction_mode(this, 2) }
+  predicate isPdbSourceFile() {
+    exists(int i |
+      file_extraction_mode(this, i) and
+      i.bitAnd(2) = 2
```

Comment:
So this is needed because the entire "mode" is now used and you need to ensure that the specific flag (in this "2" which corresponds to PDB) is present?

Reply:
Yes. And this should also be backwards compatible for old DBs.